corehq/apps/users/management/commands/populate_for_domain_for_user_history.py

```diff
@@ -0,0 +1,171 @@
+                        user_record = None
+                        # results is a list of "rows"
+                        # each entry is a dict with keys 'username', 'flag', 'row'
+                        # a 'row' is the excel row in the upload where column 'web_user' is the web user's username
+                        # a 'flag' is updated/created or the exception but if a log was created it should be
+                        # updated/created
+                        username = _get_web_users_username_by_id(user_history.user_id)
+                        if not username:
+                            raise Exception(f"Could not get username for web user {user_history.user_id}")
+                        for row in result['rows']:
+                            # a web user change via mobile user bulk upload
+                            if 'web_user' in row['row']:
+                                if row['row']['web_user'] == username:
+                                    user_record = row
+                                    break
+                            # a web user bulk upload
+                            elif row['username'] == username:
+                                user_record = row
+                                break
+                        if not user_record:
+                            raise Exception(
+                                f"Missing user {username} record in {user_history.user_upload_record_id}")
+                        if user_record['flag'] in ['created', 'updated']:
+                            for_domain = user_record['row'].get('domain') or upload_record.domain
+                        else:
+                            # raise Exception(f"Unexpected flag value {user_record['flag']}")
+                            unexpected_flag_value[user_history.pk] = user_record['flag']
+                            continue
+                    else:
+                        # for any log that has domain removal message
+                        # stay flexible with new change_messages or old message column for dry runs
+                        if hasattr(user_history, 'change_messages'):
+                            for_domain = user_history.change_messages.get(
+                                'domain', {}
+                            ).get(REMOVE_FROM_DOMAIN, {}).get('domain')
+                        # domain removal message, if present, would be the only message in user_history.message
+                        elif user_history.message and DOMAIN_REMOVE_REGEX.match(user_history.message):
+                            for_domain = DOMAIN_REMOVE_REGEX.match(user_history.message).groups()[0]
+
+                    if not for_domain and by_domain:
+                        for_domain = by_domain
+
+                if for_domain:
+                    succeeded[user_history.pk] = (getattr(user_history, BY_DOMAIN_ATTR), for_domain)
+                if for_domain:
+                    _file.write(
```

Comment:
What is this file?  Mobile worker info gets added only when this can't figure out the domain, but web users only get added when you can?

Reply:
Oh actually I see this block is executed for both mobile and web users.  I still don't follow what it's doing

Reply:
The migration would always log the updates but only actually update the records if `--save` is passed.
So, this is where we log into a file what the update was/would be.

Reply:
I don't think it'll log updates for web users where `for_domain` is falsy.

Reply:
Ya, that sounds right no?
if there is no for_domain, there wont be any update.